executor/benchmark_test.go

```diff
@@ -538,29 +545,37 @@ func benchmarkWindowExecWithCase(b *testing.B, casTest *windowTestCase) {
 		b.Fatal(err)
 	}
 
-	cols := casTest.columns()
+	decimalBuff := make([]types.MyDecimal, casTest.rows)
+	genDataFunc := func(row int, typ *types.FieldType) interface{} {
+		switch typ.Tp {
+		case mysql.TypeLong, mysql.TypeLonglong:
+			return int64(row)
+		case mysql.TypeDouble, mysql.TypeFloat:
+			return float64(row)
+		case mysql.TypeNewDecimal:
+			return decimalBuff[row].FromInt(int64(row))
```

Comment:
```suggestion
			var d types.MyDecimal
			return d.FromInt(int64(row))
```

So `decimalBuff` can be removed.